nexus/db-queries/src/db/datastore/oximeter.rs

```diff
@@ -75,6 +100,30 @@ impl DataStore {
         Ok(())
     }
 
+    /// Mark an Oximeter instance as deleted
+    ///
+    /// This method is idempotent and has no effect if called with the ID for an
+    /// already-deleted Oximeter.
+    pub async fn oximeter_delete(
+        &self,
+        opctx: &OpContext,
+        id: Uuid,
+    ) -> Result<(), Error> {
+        use db::schema::oximeter::dsl;
+
```

Comment:
It'd be nice to have authz checks on this stuff.  Not critical for this PR -- we never really went back and did this for a lot of non-external-API-driven datastore methods.

Reply:
Yeah; I copied the existing Oximeter authz checks. :sweat_smile: 

I filed #6528